File: backend/src/main/java/dev/agiro/fanel/shopping/api/ShoppingApi.java
package dev.agiro.fanel.shopping.api;

import java.util.List;
import java.util.UUID;

public interface ShoppingApi {
    List<ShoppingListDto> listLists(UUID householdId);
    ShoppingListDto getList(UUID householdId, UUID listId);
    ShoppingListDto getDefaultList(UUID householdId);
    ShoppingListDto createList(UUID householdId, String name);
    ShoppingListDto updateList(UUID householdId, UUID listId, String name);
    void deleteList(UUID householdId, UUID listId);

    ShoppingItemDto addItem(UUID householdId, UUID listId, String name, Double quantity, String unit, String category, boolean recurring);
    default ShoppingItemDto addItem(UUID householdId, UUID listId, String name) {
        return addItem(householdId, listId, name, null, null, null, false);
    }
    ShoppingItemDto updateItem(UUID householdId, UUID itemId, String name, Double quantity, String unit, String category, Boolean recurring, Boolean done);
    ShoppingItemDto setDone(UUID householdId, UUID itemId, boolean done);
    void removeItem(UUID householdId, UUID itemId);
    int clearPurchased(UUID householdId, UUID listId);
}

File: android/app/src/main/java/dev/agiro/fanel/android/data/remote/ShoppingApi.kt
package dev.agiro.fanel.android.data.remote

import retrofit2.http.Body
import retrofit2.http.DELETE
import retrofit2.http.GET
import retrofit2.http.PATCH
import retrofit2.http.POST
import retrofit2.http.PUT
import retrofit2.http.Path

interface ShoppingApi {
    @GET("/api/v1/households/{householdId}/shopping/lists")
    suspend fun listLists(@Path("householdId") householdId: String): List<ShoppingListDto>

    @POST("/api/v1/households/{householdId}/shopping/lists")
    suspend fun createList(
        @Path("householdId") householdId: String,
        @Body request: ListNameRequest
    ): ShoppingListDto

    @GET("/api/v1/households/{householdId}/shopping/lists/{listId}")
    suspend fun getList(
        @Path("householdId") householdId: String,
        @Path("listId") listId: String
    ): ShoppingListDto

    @PUT("/api/v1/households/{householdId}/shopping/lists/{listId}")
    suspend fun updateList(
        @Path("householdId") householdId: String,
        @Path("listId") listId: String,
        @Body request: ListNameRequest
    ): ShoppingListDto

    @DELETE("/api/v1/households/{householdId}/shopping/lists/{listId}")
    suspend fun deleteList(
        @Path("householdId") householdId: String,
        @Path("listId") listId: String
    )

    @GET("/api/v1/households/{householdId}/shopping/lists/default")
    suspend fun defaultList(@Path("householdId") householdId: String): ShoppingListDto

    @POST("/api/v1/households/{householdId}/shopping/lists/{listId}/items")
    suspend fun addItem(
        @Path("householdId") householdId: String,
        @Path("listId") listId: String,
        @Body request: AddItemRequest
    ): ShoppingItemDto

    @PATCH("/api/v1/households/{householdId}/shopping/items/{itemId}")
    suspend fun updateItem(
        @Path("householdId") householdId: String,
        @Path("itemId") itemId: String,
        @Body request: UpdateItemRequest
    ): ShoppingItemDto

    @DELETE("/api/v1/households/{householdId}/shopping/items/{itemId}")
    suspend fun removeItem(
        @Path("householdId") householdId: String,
        @Path("itemId") itemId: String
    )

    @POST("/api/v1/households/{householdId}/shopping/lists/{listId}/clear-purchased")
    suspend fun clearPurchased(
        @Path("householdId") householdId: String,
        @Path("listId") listId: String
    ): Map<String, Int>
}
